### PR TITLE
feat(utils): support GitLab report links

### LINKED ISSUE #722: Issues sections should link to Git URL of file at given commit with highlighted line(s)
### User story

As a user of the CLI I want to have click through experience for all actionable feedback a get from the audit issues.

At the moment the provided issues often show exactly where the error needs to get fixed, but a manual navigation to the line of code is required.

**Example:**

|Severity|Message|Source file|Line(s)|
|:--:|:--:|:--:|:--:|
|⚠️ <i>warning</i>|Lines 1-18 are not covered in any test case.|<code>examples/plugins/src/index.ts</code>|1-18|
|⚠️ <i>warning</i>|Line 104 is not covered in any test case.|<code>examples/plugins/src/lighthouse/src/lighthouse.plugin.ts</code>|104|

|Severity|Message|Source file|Line(s)|
|:--:|:--:|:--:|:--:|
|🚨 <i>error</i>|1st branch is not taken in any test case.|<code>examples/plugins/src/file-size/src/file-size.plugin.ts</code>|101|
|🚨 <i>error</i>|1st branch is not taken in any test case.|<code>examples/plugins/src/index.ts</code>|1|

**Problems to be solved:**
- git repo could be hosted in GitLab/GitHub/Azure/Bitbucket...
- each Git provider has their own URL structure
- git remove -v should give us origin
- how to detect provider from self-hosted repos? (common for enterprises)
- if we cannot detect provider, then we shouldn't include the link

### Acceptance criteria

- [ ] supported hosting provider:
  - GitLab
  - GitHub
- [ ] provide config `remoteUrl` to serve users with self-hosted repos. (common for enterprises)
- [ ] if we cannot detect provider, then we shouldn't include the link

**Example:**

|Severity|Message|Source file|Line(s)|
|:--:|:--:|:--:|:--:|
|⚠️ <i>warning</i>|Lines 1-18 are not covered in any test case.| <a href="https://github.com/code-pushup/cli/blob/main/examples/plugins/src/index.ts#L1-L18"><code>examples/plugins/src/index.ts</code></a> |1-18|
|⚠️ <i>warning</i>|Line 104 is not covered in any test case.| <a href="https://github.com/code-pushup/cli/blob/main/examples/plugins/src/lighthouse/src/lighthouse.plugin.ts#L104"><code>examples/plugins/src/lighthouse/src/lighthouse.plugin.ts</code></a> |104 |

| Severity|Message|Source file|Line(s) |
|:--:|:--:|:--:|:--:|
|🚨 <i>error</i>|1st branch is not taken in any test case.|<a href="https://github.com/code-pushup/cli/blob/main/examples/plugins/src/file-size/src/file-size.plugin.ts#L101"><code>examples/plugins/src/file-size/src/file-size.plugin.ts</code></a>|101|
|🚨 <i>error</i>|1st branch is not taken in any test case.|<a href="https://github.com/code-pushup/cli/blob/main/examples/plugins/src/index.ts#L1"><code>examples/plugins/src/index.ts</code></a>|1|

### Implementation details

Get remote from current git: `git remote` -> 'origin'

Get remote url from current git: `git remote get-url origin` -> 'git@github.com:code-pushup/cli.git'

```ts
function generateGitHubUrl(source: Pick<SourceFileLocation, 'file' | 'position'>, remoteUrl: string, branch: string): string {
  const {file, position} = source;
  const {startLine = 0, endLine = 0} = position ?? {};
  const repoPath = remoteUrl.match(/github\.com[:/](.+)\.git/)?.[1];
  const lines = startLine ? `#L${startLine}${endLine ? `-L${endLine}` : ''}` : '';
  return `https://github.com/${repoPath}/blob/${branch}/${file}${lines}`;
}

function generateGitLabUrl(source: Pick<SourceFileLocation, 'file' | 'position'>, remoteUrl: string, branch: string): string {
  const {file, position} = source;
  const {startLine = 0, endLine = 0} = position ?? {};
  const repoPath = remoteUrl.match(/gitlab\.com[:/](.+)\.git/)?.[1];
  const lines = startLine ? `#L${startLine}${endLine ? `-${endLine}` : ''}` : '';
  return `https://gitlab.com/${repoPath}/-/blob/${branch}/${file}${lines}`;
}

```

**Related Issues:**

https://github.com/code-pushup/cli/issues/149

### PR DESCRIPTION
**Pull request by hanna-skryl**
Closes #722 

This PR adds support for generating GitLab links in markdown reports when running in GitLab CI/CD, using predefined environment variables. It also refactors the environment detection logic to use a centralized `environmentChecks` record.

### CODE REVIEW

File: packages/utils/src/lib/reports/environment-type.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+const environmentChecks: Record<EnvironmentType, () => boolean> = {
+  vscode: () => process.env['TERM_PROGRAM'] === 'vscode',
+  github: () => process.env['GITHUB_ACTIONS'] === 'true',
+  gitlab: () => process.env['GITLAB_CI'] === 'true',
+  other: () => true,
+};
+
+export function getEnvironmentType(): EnvironmentType {
+  return (
+    SUPPORTED_ENVIRONMENTS.find(env => environmentChecks[env]()) ?? 'other'
```
Comment (BioPhoton) [inline packages/utils/src/lib/reports/environment-type.ts:12]: A unit-test would be nice here

**Review by BioPhoton — CHANGES_REQUESTED** [review]
Nice!! Left a small comment.

**Review by matejchalk — APPROVED** [review]
LGTM :+1:

File: packages/utils/src/lib/reports/environment-type.ts (hunk 1 above)
Comment (hanna-skryl) [reply packages/utils/src/lib/reports/environment-type.ts:12]: Done!

**Review by matejchalk — APPROVED** [review]

**Review by vmasek — APPROVED** [review]

**Pull request merged**